### PR TITLE
Add `mill` extension to supported ones

### PR DESCRIPTION
**Pull request by lolgab**

### CODE REVIEW

**Review by tgodzik — CHANGES_REQUESTED** [review]
Looks good! You will just need to run scalafmt

**Review by kitbellew — CHANGES_REQUESTED** [review]

**Review by tgodzik — APPROVED** [review]
LGTM!

File: semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SemanticdbPipeline.scala [hunk 1]
```diff
@@ -19,10 +19,14 @@ trait SemanticdbPipeline extends SemanticdbOps {
...
-        fileName.endsWith(".scala") || fileName.endsWith(".sc") || fileName.endsWith(".java")
+        val lastDotIndex = fileName.lastIndexOf('.')
+        lastDotIndex > 0 && {
+          val ext = fileName.substring(lastDotIndex + 1)
+          SemanticdbPipeline.supportedExtensions.contains(ext)
+        }
       }
-      val matchesFilter = Option(unit.source.file).flatMap(f => Option(f.file))
-        .map(f => config.fileFilter.matches(f.getAbsolutePath)).getOrElse(true)
+      val matchesFilter = Option(unit.source.file)
```
Comment (kitbellew) [inline semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SemanticdbPipeline.scala:28]: it's still a val and hence will be evaluated every time, including when `!matchesExtension`

**Review by kitbellew — APPROVED** [review]
thank you!

**Pull request merged**